enterprise/frontend/src/embedding-sdk/cli/utils/ask-for-db-connection-info.ts

```diff
@@ -38,7 +38,7 @@ export async function askForDatabaseConnectionInfo(options: Options) {
     const shouldShowField =
       !visibleIf ||
       Object.entries(visibleIf).every(
-        ([key, expected]) => connection[key] === expected,
+        ([key, expected]) => !!connection[key] === !!expected,
```

Comment:
Are you sure `visible-if` is what we should use here? It seems `visible-if` could also be a string 
![image](https://github.com/user-attachments/assets/8db928c7-f1a7-4ca2-9397-9f41ae076d99)

So casting it to a boolean looks off to me.

From looking at [the test data](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/databases/components/DatabaseForm/tests/setup.tsx), it seems `visible-if` is used to display other fields based on the current value of existing fields for a database. That means we probably won't need to have `shouldShowField` at all.

I tested by removing `shouldShowField` entirely, and no additional fields are displayed.

Reply:
@WiNloSt From what I've tested, `visible-if` is actually needed for other database engines that has choices, specifically Snowflake and MongoDB. Snowflake has `use-hostname` and MongoDB has `use-conn-uri` that conditionally provides different configuration fields. Let me test Snowflake and MongoDB on master as you've merged the PR.

```
const askSectionChoice = async (field: EngineField) => {
  // Snowflake allows to connect with either hostname or account name.
  if (field.name === "use-hostname") {
    const choice = await select({
      message: "Do you want to connect with hostname or account name?",
      choices: [
        { name: "Hostname", value: "hostname" },
        { name: "Account name", value: "account" },
      ],
    });

    return choice === "hostname";
  }

  // MongoDB allows to connect with either hostname or connection string.
  if (field.name === "use-conn-uri") {
    const choice = await select({
      message: "Do you want to connect with hostname or connection string?",
      choices: [
        { name: "Hostname", value: "hostname" },
        { name: "Connection String", value: "conn-uri" },
      ],
    });

    return choice === "conn-uri";
  }
};
```

cc @oisincoveney

Reply:
that makes sense it seems we need to bring `visible-if` back, but casting it to boolean shouldn't be the fix.

Reply:
I have two things there:
- maybe for the askSectionChoice function we explicitly identify each block as being a a part of the database instead of using a proxy value - instead of `field.name === use-hostname` we use `db_name === snowflake` or something to that effect
- maybe we write unit tests for this one? both you and kelvin were both saying "in my tests" so we may as well create some objective tests lol

but otherwise it makes sense. the reason I was trying things with a `!!` here was that were differentiating between having a truthy value and having a falsy value

Reply:
@oisincoveney ah, the reason I went for `field.name` is because the (section) fields from the database settings are already ordered, so it asks for these options in the same order as we have in the UI. testing this sounds like a good idea too.

Reply:
ah ok sounds good! I was sure you had a good reason for that as well 😄

Reply:
@WiNloSt @oisincoveney I figured it out, the `visible-if` for password requires `use-auth-provider` to be false, but we haven't set that. The original logic for `===` is already correct. The password field works now, and in addition we support the "auth providers" settings too.